.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish 📦 to PyPi 🚀
+
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  build-n-publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.10.0
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.10.0
+
+    - name: Gets the tag version
+      id: context
+      run: |
+        echo ::set-output name=TAG_VERSION::${GITHUB_REF#refs/tags/}
+
+    - name: Setup the Python Environment by installing Poetry
+      uses: ./.github/actions/setup-python-build-env
+
+    - name: Poetry bump version, build and publish
+      shell: bash
+      run: |
+        poetry version $TAG_VERSION
```

Comment:
TIL about `poetry version`.

Reply:
Well, I think it is easier if the version comes from the released tag, however I see the gap that this causes in respect to the pyproject.toml. Ideally, the CI job would release the tagged version and update the pyproject.toml, but I couldnt find a way to do this... What we can possibly do is to check if the tagged version matches the version in pyproject.toml and if it does not, aborts the upload, what do you think?

>  Note that update CHANGELOG.md and updating __version__ in doc/source/conf.py is also part of a release.
That is right...not sure how to proceed refarding the CHANGELOG.md, is there any check we can do? This is one of the pieces, that I guess, have to be done manually, right?

Reply:
> What we can possibly do is to check if the tagged version matches the version in pyproject.toml and if it does not, aborts the upload, what do you think?

I like that solution. You could run `poetry version <version>` and check if the repo is dirty afterwards. If not, continue to upload the project. If the repo is dirty, abort.

Reply:
check my changes, please

Reply:
During my edition before, my answer/question got caught by the quote `>`

> Note that update CHANGELOG.md and updating version in doc/source/conf.py is also part of a release.

That is right...not sure how to proceed refarding the CHANGELOG.md, is there any check we can do? This is one of the pieces, that I guess, have to be done manually, right?

Reply:
> .not sure how to proceed regarding the CHANGELOG.md, is there any check we can do?

I think there are tools around to generate a CHANGELOG. But that seem a bit overkill IMHO. Maybe you can print a reminder when one runs `make release` saying: 'Don't forget to update CHANGELOG.md'.

Reply:
all right. We can try to cover that in another PR as I was thinking on adding also isort and black for formatting the code, so we could add that together

Reply:
> adding also isort and black

Yes, please!

Reply:
Check my code, please. I was able to add the CURRENT_VERSION as a shell variable into the target but I couldnt find a way then to expand it into the sed command, so I had to add the variable at the top of the Makefile